internal/testing/s3_test.go

```diff
@@ -180,37 +182,49 @@ func TestS3List(t *testing.T) {
 		}
 	}
 
+	flatten := func(res <-chan minio.ObjectInfo) []string {
+		var objs []string
+		for obj := range res {
+			if obj.Err != nil {
+				t.Fatal(err)
+			}
+			objs = append(objs, obj.Key)
+		}
+		return objs
+	}
+
 	tests := []struct {
 		prefix string
-		result []string
+		want   []string
 	}{
 		{
 			prefix: "",
-			result: []string{
-				"sample.jpg",
-				"photos/",
-				"photos2/",
-			},
+			want:   []string{"/FOO/", "/fileś/", "/foo/", "/gab/"},
```

Comment:
I'm not sure this is correct. Looking at the links from the comment above, S3 will search for the first occurrence of a delimiter which is `/` in our case and make the substring up to the first occurrence the common prefix.

So imo this would only return `/` as the common prefix.

Maybe the best thing to do would actually be trying it against AWS S3 or Minio to verify our test behaves the same way.